src/main/java/org/carlmontrobotics/robotcode2023/commands/RunRoller.java

```diff
@@ -12,31 +11,32 @@
 public class RunRoller extends CommandBase {
 
     private final Roller roller;
-    private final Color ledColor;
     private final Timer timer = new Timer();
     private final RollerMode mode;
 
-    public RunRoller(Roller roller, RollerMode mode, Color ledColor) {
+    public RunRoller(Roller roller, RollerMode mode) {
         addRequirements(this.roller = roller);
         this.mode = mode;
-        this.ledColor = ledColor;
     }
 
     @Override
     public void initialize() {
-        roller.setSpeed(mode.speed);
-        roller.setLedColor(ledColor);
+        if (mode.speed > 0) { // should not interrupt command to stop rollers
+            if(roller.hasGamePiece() && isIntake()) cancel();
+            if(!roller.hasGamePiece() && !isIntake()) cancel(); 
+        }
         timer.reset();
-
-        //if(roller.hasGamePiece() == mode.intake) cancel();
+        roller.setRollerMode(mode);
     }
 
     @Override
     public void execute() {}
 
     @Override
     public void end(boolean interrupted) {
-        roller.setSpeed(0);
+        // keep it running if its a cone
+        if (mode.obj != GameObject.CONE)
+            roller.setSpeed(0);
```

Comment:
This is not sufficient to keep the roller running. Think about what will happen when the `RollerCommand` ends.